client/checkout/platform-checkout/email-input-iframe.js

```diff
@@ -19,13 +20,36 @@ export const handlePlatformCheckoutEmailInput = ( field, api ) => {
 	const parentDiv = platformCheckoutEmailInput.parentNode;
 	spinner.classList.add( 'wc-block-components-spinner' );
 
-	// Make the iframe wrapper.
+	// Make the login session iframe wrapper.
+	const loginSessionIframeWrapper = document.createElement( 'div' );
+	loginSessionIframeWrapper.setAttribute( 'role', 'dialog' );
+	loginSessionIframeWrapper.setAttribute( 'aria-modal', 'true' );
+	loginSessionIframeWrapper.classList.add(
+		'platform-checkout-login-session-iframe-wrapper'
+	);
```

Comment:
This wrapper will block the entire checkout screen for _all_ users regardless of WooPay account status: this seems like it might get a little confusing for non-WooPay users, since the entire checkout will be blocked indefinitely until the request to check for an active session is completed. Is there any reason why we need to block the whole checkout while this request is being made or can we not wait until the redirect response returns `true` to bring the curtains down?

Reply:
Probably we can ask for a design decision on this. It is absolutely true that this experience would be confusing for the non-WooPay user. My initial thought on this was if the page is not blocked and the user is a logged in WooPay user, they can start editing the fields on the checkout page and in the middle of their edit suddenly the redirection will start :sweat_smile:

Reply:
Cool, I don't really have a huge issue with this approach, but would be nice to get some design feedback on it! 

In my humble opinion, I wouldn't mind that a WooPay user starts filling in their details before being interrupted and whisked away, because they would only be sitting and waiting in the checkout anyways and at least they wouldn't have to fill in this information twice. However, forcing a non-WooPay user to sit and wait for the request is a little bit of a waste of their time, since there will be no reward received by them for this time spent.

But what do I know about UX/UI? (Answer: not a lot. 😁)

Reply:
I had a chat with @rogermattic and decided to go for the approach you suggested to unblock most of the users. Fixed in [8aa1f0d](https://github.com/Automattic/woocommerce-payments/pull/4328/commits/8aa1f0db9f8e22080527802affd2856c8e1ae8e0), now we are not blocking the user from editing the checkout fields while the request to check for a login session is in progress.